ops/services/app_service/main.tf

```diff
@@ -50,6 +50,46 @@ resource "azurerm_app_service" "service" {
     }
   }
 
+  lifecycle {
+    ignore_changes = [
+      site_config[0].linux_fx_version
+    ]
+  }
+}
+
+resource "azurerm_app_service_slot" "staging" {
+  name                = "staging"
+  app_service_name    = azurerm_app_service.service.name
+  app_service_plan_id = azurerm_app_service_plan.service_plan.id
+  resource_group_name = var.resource_group_name
+  location            = var.resource_group_location
+
+  app_settings = merge(var.app_settings, {
+    "DOCKER_REGISTRY_SERVER_PASSWORD"                = data.terraform_remote_state.global.outputs.acr_simeplereport_admin_password
+    "DOCKER_REGISTRY_SERVER_URL"                     = "https://${data.terraform_remote_state.global.outputs.acr_simeplereport_name}.azurecr.io"
+    "DOCKER_REGISTRY_SERVER_USERNAME"                = data.terraform_remote_state.global.outputs.acr_simeplereport_name
+    "WEBSITES_PORT"                                  = "8080"
+    "WEBSITE_DNS_SERVER"                             = "168.63.129.16"
```

Comment:
I confess to having no idea where it comes from (wooo copy pasta!) but I agree it should be _someplace_ that is not a random parameter. Honestly I am still trying to figure out what env variables app-services wants, needs, and/or provides.

Reply:
It looks like it comes from here: https://docs.microsoft.com/en-us/azure/app-service/web-sites-integrate-with-vnet#azure-dns-private-zones.

I just noticed this was in `services` though, so I guess hard-coding it here is fine since it'd be the only place it'd show up (in theory).

Reply:
I'm actually not 100% sure we need it at all in that case but I should probably test that assumption. 😁 Thanks for tracking it down!

Reply:
We do need it. We're doing the exact same thing on ReportStream. I have no idea why Azure doesn't set the correct DNS automatically, but if that's not in the config the app isn't able to lookup private IPs via DNS.